torch/optim/lr_scheduler.py

```diff
@@ -190,14 +201,21 @@ def __init__(self, optimizer, step_size, gamma=0.1, last_epoch=-1):
         super(StepLR, self).__init__(optimizer, last_epoch)
 
     def get_lr(self):
+        if (self.last_epoch == 0) or (self.last_epoch % self.step_size != 0):
+            return [group['lr'] for group in self.optimizer.param_groups]
+        return [group['lr'] * self.gamma
+                for group in self.optimizer.param_groups]
+
+    def legacy_get_lr(self):
```

Comment:
IMO, the name `legacy_get_lr` is pretty unclear.  It would be better to name it something that makes it clearer that it computes a closed form based on `last_epoch` and the `base_lrs`.  Maybe also consider taking `last_epoch` explicitly as an argument.